### PR TITLE
Model API Update

### PR DESCRIPTION
**Pull request by paulguerrie**
# Description

Updating the model APIs for a better developer experience.

## Type of change

Please delete options that are not relevant.

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [X] New feature (non-breaking change which adds functionality)
-   [ ] This change requires a documentation update

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -142,57 +154,46 @@ params = {
...
 ```python
-from inference.core.data_models import ObjectDetectionInferenceRequest
-from inference.models.yolov5.yolov5_object_detection import (
-    YOLOv5ObjectDetectionOnnxRoboflowInferenceModel,
-)
+from inference.models.utils import get_roboflow_model
 
-model = YOLOv5ObjectDetectionOnnxRoboflowInferenceModel(
-    model_id="soccer-players-5fuqs/1", device_id="my-pc", 
+model = get_roboflow_model(
```
Comment (capjamesg) [inline README.md:165]: Could you also add CLIP and SAM examples here too (assuming they are 2-3 lines of code each like these ones)? Maybe in separate example code blocks like the supervision README.

File: README.md [hunk 2]
```diff
@@ -142,57 +154,46 @@ params = {
 res = requests.post(url, params=params)
 print(res.json())
 ```
+
 **Pip Quickstart**:
```
Comment (capjamesg) [inline README.md:158]: `Pip` should be `pip` (lowercase `p`).

File: README.md [hunk 3]
```diff
@@ -94,19 +106,19 @@ docker run --network=host --gpus=all roboflow/roboflow-inference-server-gpu:late
...
-  
+
 - Run on Nvidia Jetson with JetPack `4.x`:
 
 ```bash
 docker run --privileged --net=host --runtime=nvidia roboflow/roboflow-inference-server-trt-jetson:latest
 ```
-  
+
 - Run on Nvidia Jetson with JetPack `5.x`:
```
Comment (capjamesg) [inline README.md:122]: `NVIDIA` should be all in capital letters.

File: README.md [hunk 4]
```diff
@@ -94,19 +106,19 @@ docker run --network=host --gpus=all roboflow/roboflow-inference-server-gpu:late
...
-  
+
 - Run on Nvidia GPU with TensorRT Runtime:
 
 ```bash
 docker run --network=host --gpus=all roboflow/roboflow-inference-server-trt:latest
 ```
-  
+
 - Run on Nvidia Jetson with JetPack `4.x`:
```
Comment (capjamesg) [inline README.md:116]: NVIDIA should be all in capital letters.

File: inference/models/__init__.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
+try:
```
Comment (capjamesg) [inline inference/models/__init__.py:1]: What is happening here?

**Review by capjamesg — COMMENTED** [review]
Overall looks good to me. Given the regression tests are passing per your comment in Slack (`The fact that the regression tests all pass tells me I didn't break anything critical.`), this looks good. I have left a couple of comments for you to review.

File: inference/models/__init__.py (hunk 5 above)
Comment (paulguerrie) [reply inference/models/__init__.py:1]: `clip` `sam` and `gaze` are all optional models so a user might not have the dependancies needed to import them.

File: README.md (hunk 1 above)
Comment (paulguerrie) [reply README.md:165]: Yes, I'll do that in a "soon to follow" PR where I also add extras to our pip install for clip and sam (by replacing their github repos in our requirements files with the roboflow hosted pypi versions)

File: README.md (hunk 2 above)
Comment (paulguerrie) [reply README.md]: Done

File: README.md (hunk 3 above)
Comment (paulguerrie) [reply README.md]: Done

File: inference/models/__init__.py (hunk 5 above)
Comment (capjamesg) [reply inference/models/__init__.py:1]: Sounds good!

**Review by capjamesg — APPROVED** [review]
Looks good!

**Pull request merged**